core/src/helpers.ts

```diff
@@ -168,6 +172,80 @@ const buildRestList = <Model>(
   });
 };
 
+type TCreateSpecializedTransformersParams<TModel> = {
+  type: 'rest' | 'graphql';
+  buildFields?: (keyof TModel)[];
+};
+const createSpecializedTransformers = <TModel>({
+  type,
+  buildFields,
+}: TCreateSpecializedTransformersParams<TModel>) => {
+  return {
+    [type]: Transformer<TModel, TModel>(type, {
+      buildFields: buildFields,
+    }),
+  };
+};
+
+type TCreateSpecializedBuilderParams<TModel> = {
+  modelFieldsConfig: TModelFieldsConfig<TModel>;
+  type: 'rest' | 'graphql';
+  name: string;
+};
+const createSpecializedBuilder = <TModel>(
+  params: TCreateSpecializedBuilderParams<TModel>
+) => {
+  const modelBuilder = Builder<TModel>({
+    type: params.type,
+    generator: Generator<TModel>({
+      fields: params.modelFieldsConfig.fields,
+    }),
+    name: params.name,
+    transformers: createSpecializedTransformers<TModel>({
+      type: params.type,
+    }),
+    postBuild: params.modelFieldsConfig.postBuild,
+  });
+
+  return modelBuilder as TBuilder<TModel>;
+};
+
+const createCompatibilityBuilder = <TModel>(params: {
```

Comment:
Helper to create the compatibility builder so current data models can be migrated to the new patterns but still (temporary) expose the same external API (backwards compatibility).